hub/core/dataset/dataset.py

```diff
@@ -2222,8 +2222,13 @@ def _save_view_in_path(
         num_workers: int,
         **ds_args,
     ):
-        """Saves this view at a given dataset path"""
-        vds = hub.dataset(path, **ds_args)
+        """Stores this view at a given dataset path"""
+        if path == self.path:
```

Comment:
path and self.path can be off by a trailing slash (`/`) or one can be a relative path and the other can be an absolute path, both resolving to the same location on disk. So for local paths you have to normalize both with `os.path.absath` before comparison.